src/cascadia/Remoting/Monarch.cpp

```diff
@@ -762,4 +762,32 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                               TraceLoggingKeyword(TIL_KEYWORD_TRACE));
         }
     }
+
+    Windows::Foundation::Collections::IMap<uint64_t, winrt::hstring> Monarch::GetPeasantNames()
+    {
+        auto names = winrt::single_threaded_map<uint64_t, winrt::hstring>();
+
+        std::vector<uint64_t> peasantsToErase{};
+        for (const auto& [id, p] : _peasants)
+        {
+            try
+            {
+                names.Insert(id, p.WindowName());
+            }
+            catch (...)
+            {
+                LOG_CAUGHT_EXCEPTION();
+                peasantsToErase.push_back(id);
+            }
+        }
+
+        // Remove the dead peasants we came across while iterating.
+        for (const auto& id : peasantsToErase)
+        {
+            _peasants.erase(id);
+            _clearOldMruEntries(id);
+        }

```

Comment:
This feels a little weird to do in `GetPeasantNames`. Do we do this in other parts of the `Monarch`? Is this just a responsibility where whenever we get the chance to clean up, we should?

Reply:
It looks like anywhere inside `Monarch` that iterates over the whole list of Peasants has a part where it will do some cleanup of dead Peasants. I'm guessing since there really isn't a good way to "automate" the cleanup of dead peasants, it'll just fall upon the Monarch to do it when browsing the whole list.